code/modules/projectiles/guns/energy/experimental.dm

```diff
@@ -48,3 +50,122 @@
+#define SPUR_NO_POWER 0
+
+/obj/item/weapon/gun/energy/spur
+	name = "\improper Spur"
+	desc = "A masterpiece crafted by the legendary gunsmith of a far-away planet."
+	icon = 'icons/obj/gun_experimental.dmi'
+	icon_state = "spur"
+	item_state = null
+	origin_tech = null
+	projectile_type = "/obj/item/projectile/spur"
+	charge_cost = 100
+	cell_type = "/obj/item/weapon/cell"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guns_experimental.dmi', "right_hand" = 'icons/mob/in-hand/right/guns_experimental.dmi')
+	fire_delay = 0
+	recoil = 1
+	var/charge_tick = 0
+	var/firelevel = SPUR_FULL_POWER
+
+/obj/item/weapon/gun/energy/spur/New()
+	..()
+	processing_objects.Add(src)
+
+
+/obj/item/weapon/gun/energy/spur/Destroy()
+	processing_objects.Remove(src)
+	..()
+
+/obj/item/weapon/gun/energy/spur/process()
+	charge_tick++
+	if(charge_tick < 2) return 0
+	charge_tick = 0
+	if(!power_supply) return 0
+	power_supply.give(100)
+	levelChange()
+	return 1
+
+/obj/item/weapon/gun/energy/spur/afterattack(atom/A as mob|obj|turf|area, mob/living/user as mob|obj, flag, params, struggle = 0)
+	levelChange()
+	..()
+
+/obj/item/weapon/gun/energy/spur/proc/levelChange()
+	var/maxlevel = power_supply.maxcharge
+	var/level = power_supply.charge
+	var/newlevel = 0
+	if(level == maxlevel)
+		newlevel = SPUR_FULL_POWER
+	else if(level >= ((maxlevel/3)*2))
+		newlevel = SPUR_HIGH_POWER
+	else if(level >= (maxlevel/3))
+		newlevel = SPUR_MEDIUM_POWER
+	else if(level >= charge_cost)
+		newlevel = SPUR_LOW_POWER
+	else
+		newlevel = SPUR_NO_POWER
+
+	if(firelevel >= newlevel)
+		firelevel = newlevel
+		set_firesound()
+		return
+
+	firelevel = newlevel
+	set_firesound()
+	var/levelupsound = null
+	switch(firelevel)
+		if(SPUR_LOW_POWER)
+			levelupsound = 'sound/weapons/spur_chargelow.ogg'
+		if(SPUR_MEDIUM_POWER)
+			levelupsound = 'sound/weapons/spur_chargemed.ogg'
+		if(SPUR_HIGH_POWER)
+			levelupsound = 'sound/weapons/spur_chargehigh.ogg'
+		if(SPUR_FULL_POWER)
+			levelupsound = 'sound/weapons/spur_chargefull.ogg'
+
+	if(levelupsound)
+		for(var/mob/M in get_turf(src))
+			M.playsound_local(M, levelupsound, 100, 0)
+			spawn(1)
+				M.playsound_local(M, levelupsound, 75, 0)
+
+
+/obj/item/weapon/gun/energy/spur/proc/set_firesound()
+	switch(firelevel)
+		if(SPUR_HIGH_POWER,SPUR_FULL_POWER)
+			fire_sound = 'sound/weapons/spur_high.ogg'
+			recoil = 1
+		if(SPUR_MEDIUM_POWER)
+			fire_sound = 'sound/weapons/spur_medium.ogg'
+			recoil = 0
+		if(SPUR_LOW_POWER,SPUR_NO_POWER)
+			fire_sound = 'sound/weapons/spur_low.ogg'
+			recoil = 0
+	return
+
+/obj/item/weapon/gun/energy/spur/update_icon()
+	return
+
+/obj/item/weapon/gun/energy/spur/polarstar
+	name = "\improper Polar Star"
+	desc = "Despite being incomplete, the severe wear on this gun shows to which extent it's been used already."
+	icon_state = "polarstar"
+	fire_delay = 1
+
+/obj/item/weapon/gun/energy/spur/polarstar/process()
+	charge_tick++
+	if(charge_tick < 2) return 0
+	charge_tick = 0
```

Comment:
I was going to say "so that LevelChange() is still called regularly when the gun is recharged" but then I remember that LevelChange() is called anyway when you fire the gun, I'm dumb.